app/javascript/components/tooltips/Concept.tsx

```diff
@@ -0,0 +1,244 @@
+import React, { useRef, useEffect, useReducer } from 'react'
+import { usePopper } from 'react-popper'
+import { useQuery } from 'react-query'
+import { Loading } from '../common/Loading'
+
+type ShowState = {
+  requestHover: boolean
+  requestFocus: boolean
+  hover: boolean
+  focus: boolean
+  show: boolean
+}
+
+type ShowActionSource =
+  | 'force-close'
+  | 'request-hover'
+  | 'request-focus'
+  | 'hover'
+  | 'focus'
+
+type ShowActionValue = 'show' | 'hide'
+
+type OpenTooltip = [string, React.Dispatch<unknown>]
+
+/**
+ * OpenTooltips maintains an array of open tooltips
+ * Only 1 should be active at once, but given the chance of race conditions
+ * A array is used to track potentially multiple open
+ */
+
+const openTooltips: OpenTooltip[] = []
+
+const indexOfOpenTooltip = (id: string) =>
+  openTooltips.findIndex(([openId]) => openId === id)
+
+const hasOpenTooltip = (id: string) =>
+  indexOfOpenTooltip(id) === -1
+    ? openTooltips.length > 0
+    : openTooltips.length > 1
+
+const addOpenTooltip = (id: string, dispatcher: React.Dispatch<unknown>) => {
+  if (openTooltips.find(([openId]) => openId === id)) {
+    return
+  }
+
+  openTooltips.push([id, dispatcher])
+}
+
+const removeOpenTooltip = (id: string) => {
+  const index = indexOfOpenTooltip(id)
+  if (index !== -1) {
+    openTooltips.splice(index, 1)
+  }
+}
+
+const closeOtherOpenTooltips = (id: string) => {
+  openTooltips.forEach(([openId, openDispatcher]) => {
+    if (openId !== id) {
+      dispatch(openId, openDispatcher, 'force-close')
+    }
+  })
+}
+
+/**
+ * showReducer
+ * This serves as a state reducer for the Concept (tooltip) Component for use with
+ * the useReducer hook.
+ */
+function showReducer(
+  state: ShowState,
+  [id = '', dispatch, source, action]: [
+    string,
+    React.Dispatch<unknown>,
+    ShowActionSource,
+    ShowActionValue
+  ]
+) {
+  const nextState: ShowState = { ...state }
+  const parsedAction = action === 'show'
```

Comment:
I found this variable name confusing.

```suggestion
  const shouldShow = action === 'show'
```